### ISSUE TITLE
Failed to watch /v1, Kind=Secret: failed to list /v1, Kind=Secret

### ISSUE THREAD

**Issue by blezoray** [external-secrets/external-secrets#2755]
Hello,

I deployed ESO on a Openshift cluster version 4.12, using OLM (Operator Lifecycle Manager).
On lab, no problem.

But on production, the "controller-manager" pod, deployed with OLM, that deploy the ESO helm chart based on the operatorconfig CR, consume huge memory (over 2 GB) and CPU.
In the logs, I can see these errors:
```
E1005 14:57:27.947872 1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.22.2/tools/cache/reflector.go:167: Failed to watch /v1, Kind=Secret: failed to list /v1, Kind=Secret: stream error when reading response body, may be caused by closed connection. Please retry. Original error: stream error: stream ID 2949; INTERNAL_ERROR
{"level":"info","ts":1696517848.088788,"logger":"helm.controller","msg":"Reconciled release","namespace":"mpms-external-secrets-operator","name":"mpms","apiVersion":"operator.external-secrets.io/v1alpha1","kind":"OperatorConfig","release":"mpms"}
I1005 14:58:31.238828 1 request.go:665] Waited for 1.097264099s due to client-side throttling, not priority and fairness, request: GET:https://172.30.0.1:443/apis/build.openshift.io/v1?timeout=32s
{"level":"info","ts":1696517918.8514946,"logger":"helm.controller","msg":"Reconciled release","namespace":"mpms-external-secrets-operator","name":"mpms","apiVersion":"operator.external-secrets.io/v1alpha1","kind":"OperatorConfig","release":"mpms"}
I1005 14:59:09.665397 1 trace.go:205] Trace[503815121]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.22.2/tools/cache/reflector.go:167 (05-Oct-2023 14:58:05.331) (total time: 64334ms):
Trace[503815121]: [1m4.334004469s] [1m4.334004469s] END
E1005 14:59:09.665426 1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.22.2/tools/cache/reflector.go:167: Failed to watch /v1, Kind=Secret: failed to list /v1, Kind=Secret: stream error when reading response body, may be caused by closed connection. Please retry. Original error: stream error: stream ID 3459; INTERNAL_ERROR
I1005 14:59:40.118865 1 request.go:665] Waited for 1.045434445s due to client-side throttling, not priority and fairness, request: GET:https://172.30.0.1:443/apis/flows.knative.dev/v1?timeout=32s
{"level":"info","ts":1696517985.8971891,"logger":"helm.controller","msg":"Reconciled release","namespace":"mpms-external-secrets-operator","name":"mpms","apiVersion":"operator.external-secrets.io/v1alpha1","kind":"OperatorConfig","release":"mpms"}
I1005 15:00:47.889320 1 request.go:665] Waited for 1.041447353s due to client-side throttling, not priority and fairness, request: GET:https://172.30.0.1:443/apis/kafka.strimzi.io/v1beta1?timeout=32s
I1005 15:00:55.460965 1 trace.go:205] Trace[1134291501]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.22.2/tools/cache/reflector.go:167 (05-Oct-2023 14:59:54.692) (total time: 60768ms):
Trace[1134291501]: [1m0.768885778s] [1m0.768885778s] END
```

The reason is we have about 20,000 secrets in the Kubernetes cluster and the controller-manager tries to get them.
This is the default behavior when controller-manager are developed using the operator SDK.

Cert-Manager has the same issue: https://github.com/cert-manager/cert-manager/issues/3229

Some info:
```
$ oc version
Client Version: 4.12.18
Kustomize Version: v4.5.7
Kubernetes Version: v1.25.12+26bab08
```
```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: external-secrets-operator
spec:
  channel: stable
  installPlanApproval: Manual
  name: external-secrets-operator
  source: community-operators
  sourceNamespace: openshift-marketplace
  startingCSV: external-secrets-operator.v0.9.5
  config:
    resources:
      requests:
        cpu: 100m
        memory: 1Gi
      limits:
        cpu: 300m
        memory: 3Gi
```

Could you investigate on that issue ?

BR.

**Comment by blezoray** [comment]
Apparently, you use an old version of the operator SDK helm image.
https://github.com/external-secrets/external-secrets-helm-operator/blob/main/Dockerfile#L2

You should upgrade it ?

**Comment by blezoray** [comment]
Hi,

I did the following test : I created a new helm controller image by replacing the From docker image to version 1.31.0
```
# Build the manager binary
FROM quay.io/operator-framework/helm-operator:v1.31.0

ENV HOME=/opt/helm
COPY watches.yaml ${HOME}/watches.yaml
COPY helm-charts  ${HOME}/helm-charts
WORKDIR ${HOME}
```
Then I directly updated the CSV (ClusterServiceVersion) on the cluster, and replaced the default image by my image.

Here are the memory consumption of the helm controller:
Before: default v0.9.5
![image](https://github.com/external-secrets/external-secrets/assets/17479277/c779e18b-db49-4210-99af-83089fb5ae99)


After: v0.9.5 with helm controller 1.31.0
![image](https://github.com/external-secrets/external-secrets/assets/17479277/a7c0eefe-9c72-490d-b6b3-8ff64a811252)

It's better but there is still a lot of memory consumption for a basic helm controller.
One reason that should explain better performance in operator-sdk 1.31.0 here: https://github.com/operator-framework/operator-sdk/pull/6354

Any advice ?

**Comment by blezoray** [comment]
I just tested the last operator-sdk/helm-controller version 1.32.0 available since this night and it fixes the issue:
![image](https://github.com/external-secrets/external-secrets/assets/17479277/a415b634-aa7c-4454-81fd-844dd1e418e3)

The 1.32.0 contains this PR: https://github.com/operator-framework/operator-sdk/pull/6377

**Comment by blezoray** [comment]
This issue has been fixed by the ESO version 0.9.6 which uses operator-sdk-helm controller version 1.32.0.

Thanks.
